objects-8/task2.js

```diff
@@ -0,0 +1,189 @@
+  {
+    id: 1,
+    name: "Предприятие 1",
+    departments: [
+      {
+        id: 2,
+        name: "Отдел тестирования",
+        employees_count: 10,
+      },
+      {
+        id: 3,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+      },
+      {
+        id: 4,
+        name: "Администрация",
+        employees_count: 15,
+      },
+    ]
+  },
+  {
+    id: 5,
+    name: "Предприятие 2",
+    departments: [
+      {
+        id: 6,
+        name: "Отдел разработки",
+        employees_count: 50,
+      },
+      {
+        id: 7,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+      },
+      {
+        id: 8,
+        name: "Отдел охраны труда",
+        employees_count: 5,
+      },
+    ]
+  },
+  {
+    id: 9,
+    name: "Предприятие 3",
+    departments: [
+      {
+        id: 10,
+        name: "Отдел аналитики",
+        employees_count: 0,
+      },
+    ]
+  }
+]
+// Задания:
+// 1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников. 
+//Для предприятия посчитать сумму всех сотрудников во всех отделах.
+function countEmployees(array){
+    let entreprise = [];
+    for (let company of array){
+        let count = 0;
+        let shortInfo = {};
+            for(let i = 0; i< company.departments.length; i++){
+                count += company.departments[i].employees_count;
+                shortInfo[company.name] = count;
+                shortInfo[company.departments[i].name] = company.departments[i].employees_count;
+
+        }
+        entreprise.push(shortInfo); 
+    }return entreprise;
+}
+
+// 2. Написать функцию, которая будет принимать 1 аргумент 
+//(id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+function showCompany(idOrName){
+  for (let company of enterprises){
+    if (company.departments.find((department) => (department.id === idOrName || department.name === idOrName))){
+      return company.name
+}
+}}
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия. Пример:addEnterprise("Название нового предприятия")
+function addEnterprise(companyName) {
+  let greatestId = Math.max(...enterprises.map(company => (company.id + company.departments.length)));
+  let companyId = greatestId + 1;
+  enterprises.push(
+    {
+      id: companyId,
+      name: companyName,
+      departments: [],
+    }
+  );
+  return enterprises 
+}
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+// Пример: addDepartment(1, "Название нового отдела")
+function addDepartment(companyId, deptName) {
+  for (let company of enterprises) {
+    let greatestId = Math.max(...enterprises.map(company => (company.id + company.departments.length)));
```

Comment:
And getNewId should be used as a function

Reply:
done